### ISSUE TITLE
Rest API response "page_size" missing in M2.4.4

### ISSUE THREAD

**Issue by anvanza** [magento/magento2#35310]
### Preconditions (*)
1. Install a 2.4.3-p1
2. 

### Steps to reproduce (*)
1. On 2.4.3-p1 create a GET request to
`rest/all/V1/orders/?searchCriteria[filterGroups][0][filters][0][field]=status&searchCriteria[filterGroups][0][filters][0][value]=processing&searchCriteria[filterGroups][0][filters][0][conditionType]=eq&searchCriteria[currentPage]=1`
2. The response should return something like:
![Screenshot 2022-04-14 at 13 56 35](https://user-images.githubusercontent.com/2848347/163386305-ebc76c09-142e-4812-a66c-0385db561e98.png)
````
{
    "items": [.....],
    "search_criteria": {
        "filter_groups": [
            {
                "filters": [
                    {
                        "field": "status",
                        "value": "processing",
                        "condition_type": "eq"
                    }
                ]
            }
        ],
        "page_size": 20,
        "current_page": 1
    },
    "total_count": 20
}
````
4. Try the same request on 2.4.4 

### Expected result (*)
1. The metadata in the response should be the exact same on Magento 2.4.3-p1 and 2.4.4

### Actual result (*)
1. The **page_size** is missing
![Screenshot 2022-04-14 at 13 56 49](https://user-images.githubusercontent.com/2848347/163385928-a4221615-ad94-4cee-a0fd-1001eb683481.png)
`````
{
    "items": [.....],
    "search_criteria": {
        "filter_groups": [
            {
                "filters": [
                    {
                        "field": "status",
                        "value": "processing",
                        "condition_type": "eq"
                    }
                ]
            }
        ],
        "current_page": 1
    },
    "total_count": 20
}
`````

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by anvanza** [comment]
@magento give me 2.4.3-p1 instance

**Comment by anvanza** [comment]
@magento give me 2.4.4 instance

**Comment by mackieee** [comment]
@anvanza Do you have a figure filled in the new Web API input limits? (Found in Configuration -> Services -> Magento Web API). 

![image](https://user-images.githubusercontent.com/883702/163387923-f841d82c-b528-43fb-a35a-f68cb82d4e74.png)

Could attempt again upon setting your own Page size default?

**Comment by anvanza** [comment]
@mackieee , I found the default page size config field. Added 20 in there but the result is still the same. The page_size is still missing in the response.

**Comment by engcom-November** [comment]
Hi @anvanza ,
Thank you for reporting and collaboration. Verified the behavior on Magento 2.4-develop branch and is reproducible. However, by setting searchCriteria[pageSize] in Params, issue is not reproducible. "page_size" retrieved in the response:
![image](https://user-images.githubusercontent.com/60198231/163415688-0a34da8b-b3c7-4e95-885e-06902ef76188.png)

As per [dev docs](https://devdocs.magento.com/guides/v2.4/rest/performing-searches.html) as well, page_size is not retrieved in the example response, if it is not set in request.

Same behavior can be observed in 2.4.3-p1 instance for "current_page" field.
Kindly recheck the behavior on Magento 2.4-develop branch by sending request with "page_size" param

**Comment by anvanza** [comment]
Hi @engcom-November ,

True, On 2.4.4 the page_size is returned when searchCriteria[pageSize] is set. 
While on 2.4.3-p1 it returned page_size even when searchCriteria[pageSize] is not set.

**Comment by engcom-November** [comment]
Hi @anvanza,

We are moving this ticket to "On Hold" as we are awaiting confirmation from PO for expected behavior.
Thank you.

**Comment by Viktorja1** [comment]
https://github.com/magento give me 2.4.4 instance

**Comment by engcom-Hotel** [comment]
Hello @anvanza,

We have got reply from the PO, and as per him this is an issue but the page size is a part of any GET rest request that is a multi-entity response. 

Even if not passed, the page size should be present, because the rest should have a fixed payload/schema.

Of course, this is not a blocker because they could include a simple logic to check for the existence of the key in the logic and just assume a default of pagesize= 20.

Hence confirming this issue for further processing.

Thanks

**Comment by GurramSravya** [comment]
@magento I am working on this

**Comment by GurramSravya** [comment]
@magento add to contributors team

**Comment by GurramSravya** [comment]
@magento I am working on this

**Comment by GurramSravya** [comment]
@magento give me 2.4.4 instance

**Comment by GurramSravya** [comment]
@magento give me 2.4.3-p1 instance

**Comment by GurramSravya** [comment]
The issue can be resolved by enabling the configuration from backend as 

Stores->Configuration->Services->Magento Web Api->Web Api Input Limits->Enable Input Limits->Default Page Size

<img width="1787" alt="image" src="https://github.com/magento/magento2/assets/136710718/37144fa0-e61a-44ee-98f5-28e4da8f7a14">
<img width="495" alt="image" src="https://github.com/magento/magento2/assets/136710718/9eeb8776-e96b-431a-bc9b-af3fb779a162">

**Comment by anvanza** [comment]
True, On 2.4.4 the page_size is returned when searchCriteria[pageSize] is set.
While on 2.4.3-p1 it returned page_size even when searchCriteria[pageSize] is not set.

Same response as in 2022...

**Comment by engcom-Delta** [comment]
Hi @anvanza ,

Thanks for your reporting and collaboration.
We have verified the issue in latest 2.4-develop instance and the issue is reproducible.
Kindly refer the screenshots.

**Steps to reproduce**
1. create a GET request to
rest/all/V1/orders/?searchCriteria[filterGroups][0][filters][0][field]=status&searchCriteria[filterGroups][0][filters][0][value]=processing&searchCriteria[filterGroups][0][filters][0][conditionType]=eq&searchCriteria[currentPage]=1
2. Observe page_size parameter is not displaying
![image](https://github.com/user-attachments/assets/3247aaa1-2f3a-41d1-b941-a0de8ef50347)
3. However, by setting searchCriteria[pageSize] in Params, issue is not reproducible.
![image](https://github.com/user-attachments/assets/cf3e0ebb-a04b-4536-a3f4-c86e20e29ecc)

Thanks.
